SUS-1195: make sure the file cache is up to date shortly after the video upload

macbre opened the pull request:
See https://wikia-inc.atlassian.net/browse/SUS-1195
## Problem

Video upload via UI pop-up ends up with an error logged to ELK:

``` json
{
    "url": "https://www.youtube.com/watch?v=tOzJHO3q_9w",
    "title": "[object] (Title: File:Https://www.youtube.com/watch?v=tOzJHO3q 9w)",
    "videoTitle": "[object] (Title: File:Dash 8 Air Iceland take off in Greenland !!)",
    "videoPageId": 435145,
    "videoProvider": null,
    "wgIsGhostVideo": null
},
```

The video is correctly uploaded, but the process cache claims that it does not exist:

```
> var_dump( wfFindFile( Title::newFromId(435145) ) )
Query mediawiki119cleanup4 (DB user: wikia_maint) (72) (slave): SELECT /* Title::newFromID CommandLineInc - 1cc0d57f-f91f-495e-a207-3c5624159e3c */  *  FROM `page`  WHERE page_id = '435145'  LIMIT 1  
bool(false)
```
## Solution

Make sure the file cache of newly uploaded video is always up to date by explicitly setting the memcache entry shortly after the successful upload,

And fix two PHP notices that make devbox users pretty annoyed :smile: 

@TK-999 / @mixth-sense / @bulaszenko

macbre committed:
SUS-1195: make sure the file cache is up to date shortly after the video upload

WikiaTech commented:
SonarQube analysis reported 19 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 11 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 5 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L7](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L7): Class "VideoFileUploader" has 29 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L44](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L44): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L87](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L87): Remove this unused "$thumbnailUrl" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L116](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L116): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L134](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L134): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L327](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L327): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L353](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L353): Remove this unused "$orgWidth" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L371](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L371): Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L466](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L466): Remove the unused function parameter "$level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L514](https://github.com/Wikia/app/blob/9ca36d26607cb65c502f45d87ae6fc06642b6e73/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L514): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)

macbre committed:
VideoFileUploader: fix a PHP notice

Notice: Only variables should be passed by reference in /usr/wikia/source/app/extensions/wikia/VideoHandlers/VideoFileUploader.class.php on line 70

WikiaTech commented:
SonarQube analysis reported 19 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 11 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 5 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 3 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L7](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L7): Class "VideoFileUploader" has 29 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L44](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L44): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L89](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L89): Remove this unused "$thumbnailUrl" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L118](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L118): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L136](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L136): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L329](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L329): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L355](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L355): Remove this unused "$orgWidth" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L373](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L373): Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L468](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L468): Remove the unused function parameter "$level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L516](https://github.com/Wikia/app/blob/72d85c38f730876c9c9a7c2f573a2b9ef1cf50a1/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L516): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)

mszabo-wikia reviewed (approved):
LGTM
[![LGTM](https://lgtm.in/p/gAosEfO7q)](https://lgtm.in/i/gAosEfO7q)

macbre committed:
SpecialVideosHelper: and one more PHP notice killed

WikiaTech commented:
SonarQube analysis reported 28 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 12 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 12 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [SpecialVideosHelper.class.php#L29](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php#L29): Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L7](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L7): Class "VideoFileUploader" has 29 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L44](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L44): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L89](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L89): Remove this unused "$thumbnailUrl" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L118](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L118): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L136](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L136): Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS112)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L329](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L329): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L355](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L355): Remove this unused "$orgWidth" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L373](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L373): Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [VideoFileUploader.class.php#L468](https://github.com/Wikia/app/blob/640e4def2b92b45a86deb03c73ce9381669bbf7f/extensions/wikia/VideoHandlers/VideoFileUploader.class.php#L468): Remove the unused function parameter "$level". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

WikiaTech commented on extensions/wikia/SpecialVideos/SpecialVideosHelper.class.php:
```diff
@@ -130,7 +130,7 @@ protected function getTotalVideos( $videoParams ) {
-		if ( $videoParams['sort'] == 'premium' ) {
+		if ( !empty( $videoParams['sort'] ) && $videoParams['sort'] == 'premium' ) {
 			$totalVideos = $mediaService->getTotalPremiumVideos();
 		} else if ( !empty( $videoParams['category'] ) ) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "category" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech replied:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "category" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
